Neos.ContentRepository.Core/Classes/Feature/NodeReferencing/Dto/NodeReferenceToWrite.php

```diff
@@ -26,32 +27,37 @@
  * (this is validated in the command handler).
  * @api used as part of commands
  */
-final readonly class NodeReferenceToWrite implements \JsonSerializable
+final readonly class NodeReferenceToWrite
 {
     public function __construct(
+        public ReferenceName $referenceName,
         public NodeAggregateId $targetNodeAggregateId,
         public ?PropertyValuesToWrite $properties
     ) {
     }
 
     /**
-     * @param array<string,mixed> $array
+     * @param array{"referenceName": string, "target": string, "properties"?: array<string, mixed>} $array
+     * @see NodeReferencesToWrite::fromArray()
      */
     public static function fromArray(array $array): self
     {
         return new self(
-            NodeAggregateId::fromString($array['targetNodeAggregateId']),
+            ReferenceName::fromString($array['referenceName']),
+            NodeAggregateId::fromString($array['target']),
             isset($array['properties']) ? PropertyValuesToWrite::fromArray($array['properties']) : null
         );
     }
 
     /**
-     * @return array<string,mixed>
+     * Provides a limited array representation which can be safely serialized in context of {@see NodeReferencesToWrite}
+     *
+     * @return array{"target": NodeAggregateId, "properties": PropertyValuesToWrite|null}
      */
-    public function jsonSerialize(): array
+    public function targetAndPropertiesToArray(): array
```

Comment:
i would rather like to inline this code to the caller instead of here?
or a generic `toArray`? Idk or make it internal :D

Reply:
yeah internal I guess... But htis might again become easier with the middle DTO.

Reply:
This has been simplified